src/sgl/func/tensor.cpp

```
@@ -28,11 +267,65 @@ Device* Tensor::device() const
+{
+    SGL_CHECK(
+        shape.size() <= kSlangPyTensorViewMaxDim,
+        "TensorView supports at most {} dimensions, got {}.",
+        kSlangPyTensorViewMaxDim,
+        shape.size()
+    );
+
+    TensorViewData data = {};
+    data.data = storage->device_address() + static_cast<uint64_t>(offset) * element_stride;
+
+    for (size_t i = 0; i < shape.size(); ++i) {
```

Cppcheck (2.20.0)

[style] The function 'insert' is never used.

(unusedFunction)